src/stapi_fastapi/routers/product_router.py

```diff
@@ -195,8 +289,52 @@ async def search_opportunities(
                 )
             case x:
                 raise AssertionError(f"Expected code to be unreachable {x}")
+
+        if prefer is Prefer.wait and self.root_router.supports_async_opportunity_search:
+            response.headers["Preference-Applied"] = "wait"
+
         return OpportunityCollection(features=features, links=links)
 
+    async def search_opportunities_async(
+        self: Self,
+        search: OpportunityPayload,
+        request: Request,
+        prefer: Prefer | None,
+    ) -> JSONResponse:
+        match await self.product._search_opportunities_async(self, search, request):
+            case Success(search_record):
+                search_record.links.append(
+                    self.root_router.opportunity_search_record_self_link(
+                        search_record, request
+                    )
+                )
+                headers = {}
+                headers["Location"] = str(
+                    self.root_router.generate_opportunity_search_record_href(
+                        request, search_record.id
+                    )
+                )
+                if prefer is not None:
+                    headers["Preference-Applied"] = "respond-async"
+                return JSONResponse(
+                    status_code=201,
+                    content=search_record.model_dump(mode="json"),
+                    headers=headers,
+                )
+            case Failure(e) if isinstance(e, ConstraintsException):
+                raise e
+            case Failure(e):
+                logger.error(
```

Comment:
why not logger.exception?

Reply:
When in Rome, I suppose. Also this: https://github.com/stapi-spec/stapi-fastapi/pull/125. 

It does seem that an exception is more correct, since we are terminating on the exception. I'll update.

Reply:
lol, yeah, I only recently learned the difference, too.